### ISSUE TITLE
Connect to Facebook

### ISSUE THREAD

**Issue by Skud** [Growstuff/growstuff#556]
Members should be able to connect their Growstuff account to their Facebook accounts.  This should use OAuth.  The resulting linked accounts should be displayed on the Growstuff member's profile.

**Comment by CloCkWeRX** [comment]
Dupe?

**Comment by cesy** [comment]
This is linking, not necessarily sign-in via Facebook. It allows someone with a Growstuff password to have facebook linked as well, which is a step beyond #645 though both could be done at once.

**Comment by CloCkWeRX** [comment]
This one is now plausible with the above PR, but probably needs a bit more discussion.

We can do the 'add facebook credential' step fairly easily - the existing authentications_controller seems to be 99% there; but we probably want to think about what kind of information we want to read (user location, photos, etc), and if there is any we want to write.

**Comment by cesy** [comment]
I think probably we don't want to read location - better to give members the choice of how granular to specify it for Growstuff separately. Reading photos is a nice feature for the long-term, as it will make adding photos to gardens much easier, but not necessary for MVP. Part of the aim of this eventually is to be able to write - adding optional "Share to Facebook" and "Share to Twitter" buttons whenever you plant or harvest something, and so on.

**Comment by maco** [comment]
Technically, we could have had share buttons ages ago. They can be done
without the API.

**Comment by cesy** [comment]
Is it worth us spinning off a separate issue for them and locking down the
design so we can get them added, then?

**Comment by CloCkWeRX** [comment]
I think it might be. Interestingly, now that #733 is "sorta working", we can probably implement this with plain links by simply rendering an appropriate og:image attribute whenever we have a more accurate picture, and populating an og:title (should already work if there's content_for :title)

**Comment by CloCkWeRX** [comment]
After #811 is bedded down, I'll pick this up and look at UI for 'connect facebook account' - should be fairly straight forward given the existing implementations.

**Comment by CloCkWeRX** [comment]
This should work now, as part of #811 - from the 'apps' member page, you can connect. There's something odd on my local around the routes which is absent in staging, so it'll need testing there + the credential setup... but yeah; should be closable shortly
